Engine/Translation/ITranslationProvider.cs

```diff
@@ -0,0 +1,173 @@
+                    {
+                        MaybeUpdateMappings();
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
+                    TapThread.Sleep(TimeSpan.FromSeconds(5));
+                }
+            });
+        });
+    }
+
+    static string AttributeValue(XElement x, string name) => x.Attributes(name).FirstOrDefault()?.Value;
+    static string ElementValue(XElement x, string name) => x.Elements(name).FirstOrDefault()?.Value;
+
+    DisplayAttribute ComputeDisplayAttribute(string key, DisplayAttribute fallback)
+    {
+        var name = GetValue(_stringLookup, $"{key}.Name") ?? fallback.Name;
+        var description = GetValue(_stringLookup, $"{key}.Description") ?? fallback.Description;
+        var group = GetValue(_stringLookup, $"{key}.Group") ?? string.Join(" \\ ", fallback.Group);
+        double order = fallback.Order;
+        if (GetValue(_stringLookup, $"{key}.Order") is { } ord)
+            if (double.TryParse(ord, out var o))
+                order = o;
+        return new DisplayAttribute(Culture, name, description, group, order, fallback.Collapsed);
+    }
+
+    DisplayAttribute ComputeDisplayAttribute(Enum e, string name)
+    {
+        var enumType = e.GetType();
+        var key = enumType.FullName + $".{name}";
+        var fallback = enumType.GetMember(name).FirstOrDefault().GetDisplayAttribute();
+        return ComputeDisplayAttribute(key, fallback);
+    }
+
+    DisplayAttribute ComputeDisplayAttribute(IReflectionData mem)
+    {
+        var key = mem switch
+        {
+            IMemberData imem => $"{imem.DeclaringType.Name}.{imem.Name}",
+            // If the type is generic, the lookup key should not include the generic type parameter
+            ITypeData td2 when td2.AsTypeData()?.Type is { } tp && tp.IsGenericType
+                => TypeData.FromType(tp.GetGenericTypeDefinition()).Name,
+            ITypeData tmem => tmem.Name,
+            _ => mem.Name
+        };
+
+        var fallback = DefaultDisplayAttribute.GetUntranslatedDisplayAttribute(mem);
+        return ComputeDisplayAttribute(key, fallback);
+    }
+
+    private ImmutableDictionary<IReflectionData, DisplayAttribute> _displayLookup = ImmutableDictionary<IReflectionData, DisplayAttribute>.Empty;
+    private ImmutableDictionary<Enum, DisplayAttribute> _enumDisplayLookup = ImmutableDictionary<Enum, DisplayAttribute>.Empty;
+    private ImmutableDictionary<string, string> _stringLookup = ImmutableDictionary<string, string>.Empty;
+
+    bool NeedsUpdate(string key)
+    {
+        if (!UpdateTime.TryGetValue(key, out var lastUpdate)) return false;
+        if (DateTime.Now - lastUpdate < TimeSpan.FromSeconds(1)) return false;
+        if (!File.Exists(key)) return false;
+        if (lastUpdate == DateTime.MinValue) return true;
+        var lastWrite = new FileInfo(key).LastWriteTime;
+        if (lastWrite > lastUpdate) return true;
```

Comment:
Actually, I would not say it is important the the last write is at a new date. If its just different from the last time we checked, that should prompt for a change.

lets encapsulate the cache validation in a function like:
```
string CacheValidationMarker(string file){
    return FileInfo(key).LastWriteTime.ToString();
}

```

We can then decide inside the function what triggers an update. It could be LastWriteTime, or LastWriteTime + Size, ...